consumer/config.go

```diff
@@ -5,39 +5,61 @@ import (
 
 	"github.com/Shopify/sarama"
 	"github.com/heetch/felice/codec"
+	"gopkg.in/retry.v1"
 )
 
 // Config is used to configure the Consumer.
 type Config struct {
 	*sarama.Config
 
+	// KafkaAddrs holds kafka brokers addresses. There must be at least
+	// one entry in the slice.
+	// Default to localhost:9092.
+	KafkaAddrs []string
+
 	// MaxRetryInterval controls the maximum length of time that
 	// the Felice consumer will wait before trying to
 	// consume a message from Kafka that failed the first time around.
-	// The default value is 5 seconds.
+	// Default to 5 seconds.
 	MaxRetryInterval time.Duration
-
-	// Codec used to decode the message key. Defaults to codec.String.
+	// Codec used to decode the message key.
+	// Default to codec.String.
 	KeyCodec codec.Codec
+
+	retryStrategy retry.Strategy
 }
 
 // NewConfig creates a config with sane defaults.
-// The Sarama Cluster group mode will always be overwritten by the consumer
-// and thus cannot be changed, as the consumer is designed to use the ConsumerModePartitions mode.
-func NewConfig(clientID string) Config {
+// Broker addresses defaults to localhost:9092.
+func NewConfig(clientID string, addrs ...string) Config {
 	var c Config
 
 	c.Config = sarama.NewConfig()
 	c.ClientID = clientID
 	c.Consumer.Return.Errors = true
 	// Specify that we are using at least Kafka v1.0
 	c.Version = sarama.V1_0_0_0
-
 	// Distribute load across instances using round robin strategy
 	c.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRoundRobin
 
 	// Felice consumer configuration
+	c.KafkaAddrs = addrs
+	if c.KafkaAddrs == nil {
```

Comment:
Just of curiosity, why it should be better?

Reply:
Because what we're really concerned about here is whether there are any addresses provided at all. If there are no addresses, I think it's appropriate to use the default, regardless of whether a nil slice is passed in or not. In Go, a nil slice is often used interchangeably with an empty slice, and checking for length rather than nilness would be acknowledge that and be slightly more defensive.